qiskit/transpiler/passes/basis/unroll_3q_or_more.py

```diff
@@ -20,6 +20,27 @@
 class Unroll3qOrMore(TransformationPass):
     """Recursively expands 3q+ gates until the circuit only contains 2q or 1q gates."""
 
+    def __init__(self, target=None, basis_gates=None):
+        """Initialize the Unroll3qOrMore pass
+
+        Args:
+            target (Target): The target object reprsenting the compilation
+                target. If specified any multiqubit instructions in the
+                circuit when the pass is run that are supported by the target
+                device will be left in place. If both this and ``basis_gates``
+                are specified only the target will be checked.
+            basis_gates (list): A list of basis gate names that the target
+                device supports. If specified any gate names in the circuit
+                which are present in this list will not be unrolled. If both
+                this and ``target`` are specified only the target will be used
+                for checking which gates are supported.
+        """
+        super().__init__()
+        self.target = target
+        self.basis_gates = None
```

Comment:
```suggestion
        self.basis_gates = set(basis_gates) if basis_gates is not None or target
```
Why we need to keep target? Seems like both define `__contains__` thus we can simplify. This doesn't block your PR. Just my curiosity.

Reply:
It's mostly leftover from an earlier local attempt. I originally wanted to use `target.instruction_supported(node.name, qargs)` to check that the multiqubit gate is defined on the qubits in the circuit. But the problem with that was when this pass is run before the layout phase we don't actually know which physical device qubits the gate will run on so we can't check that in the target.

I think longer term it makes sense to keep them separate because after https://github.com/Qiskit/qiskit-terra/pull/7807 merges we can use `instruction_supported()` to check based on gate class and potentially parameterizations too instead of just using the name.

Reply:
Thanks, this PR looks good to me.